### PR TITLE
in-progress for enabling/disabling banner and cleanup

### PR DESCRIPTION
**Pull request by songgao**
Also restructure driver related state a little bit so what's in redux store is now more "processed".

The kext popup thing only works on a fresh macOS install. Here's a screencast in vm: /keybase/team/keybase/videos/macOS-finder-integration.mov

### CODE REVIEW

**Comment by songgao** [comment]
bump!

File: shared/fs/common/try-enable-driver-on-focus.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+})
+
+type Props = {
+  appFocusedCount: number,
+  driverStatus: Types.DriverStatus,
+  onEnabled: () => void,
+  enable: () => void,
+}
+
+class Component extends React.PureComponent<Props> {
```
Comment (chrisnojima) [inline shared/fs/common/try-enable-driver-on-focus.js]: lets talk about this, i think this is likely a dangerous pattern

**Comment by songgao** [comment]
ready again!

File: shared/fs/banner/system-file-manager-integration-banner/kext-permission-popup.js [hunk 2]
```diff
@@ -36,9 +36,21 @@ const Illustration = () => (
   </Kb.Box>
 )
 
+class CancelWhenEnabled extends React.PureComponent<Props> {
+  componentDidMount() {
```
Comment (chrisnojima) [inline shared/fs/banner/system-file-manager-integration-banner/kext-permission-popup.js]: should make a helper like _cancelOnUpdate or something

**Pull request merged**